lib/http2/connection.c

```diff
@@ -1170,7 +1176,7 @@ void do_emit_writereq(h2o_http2_conn_t *conn)
     case H2O_HTTP2_CONN_STATE_OPEN:
         break;
     case H2O_HTTP2_CONN_STATE_HALF_CLOSED:
-        if (conn->num_streams.pull.half_closed + conn->num_streams.push.half_closed != 0)
+        if (conn->num_streams.pull.open + conn->num_streams.push.open != 0)
```

Comment:
Could you please explain the intent of the change? My understanding is that the connection needs to kept open until all the streams deliver their response to the client.

Reply:
Yes but we also should keep the connections which doesn't start responses yet. The lifetime of `num_streams.*.open` includes `conn->num_streams.*.half_closed`
https://github.com/h2o/h2o/blob/master/include/h2o/http2_internal.h#L432
https://github.com/h2o/h2o/blob/master/include/h2o/http2_internal.h#L445

Reply:
Thank you for the clarification. I understand now.